packages/sdk/src/mcp/agents/api.ts

```diff
@@ -52,19 +52,16 @@ export const listAgents = createApiHandler({
   name: "AGENTS_LIST",
   description: "List all agents",
   schema: z.object({}),
+  async canAccess(_, c) {
+    return await canAccessWorkspaceResource(this.name, c);
+  },
```

Comment:
where `this.name` comes from? looks a bit implicit

perhaps you can set a default can access when creating an createApiHandler?

Reply:
this.name comes from the APIHandlerDefinition. IMO, this.name is easy to get that it comes from the name defintion from the object.